card/mod/02_basic_types/set/type/pointer.rb

```diff
@@ -1,27 +1,27 @@
 
 
-event :add_and_drop_items, before: :approve, on: :save do
+event :add_and_drop_items, :prepare_to_validate, on: :save do
   adds = Env.params['add_item']
   drops = Env.params['drop_item']
   Array.wrap(adds).each { |i| add_item i } if adds
   Array.wrap(drops).each { |i| drop_item i } if drops
 end
 
-event :insert_item_event, before: :approve, on: :save, when: proc {|c| Env.params['insert_item']} do
+event :insert_item_event, :prepare_to_validate, on: :save, when: proc {|c| Env.params['insert_item']} do
```

Comment:
Unused block argument - `c`. You can omit the argument if you don't care about it.<br>Line is too long. [104/80]<br>Space between { and | missing.<br>Space missing inside }.